### DISCUSSION TITLE
xmake show -l modes is broken

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#1710] [General]
https://xmake.io/#/plugin/builtin_plugins?id=show-builtin-compilation-modes-list

![image](https://user-images.githubusercontent.com/42310255/134792827-ad13d93f-2950-4a2c-8f40-d39884d43118.png)

modes只在`xmake show -l rules`里面才展示，`xmake show -l modes`挂了；另外xmake show -l rules一行宽度有点太宽了，看起来排版挺乱的，有没有办法根据当前console的字符宽度调整一下，类似ls命令的输出？

**Comment by waruqi** [comment]
有  xmake show -l buildmodes

宽度自适应是可以做，就是要费点时间，暂时我先调整到 一列三行了，再试试，应该会好点。

**Reply by xq114** [reply]
明白了，是接口改了，文档还没更新

**Reply by waruqi** [reply]
> 明白了，是接口改了，文档还没更新

文档有些地方难免会有遗漏 忘记同步了，也可以帮忙提 pr 更新下。。目前都是发版前根据 changelog 更新一波文档，但是一些小patch ，可能没在 changelog 里面记录，之后我就会忘记掉。。
